### ISSUE TITLE
Cannot reimport instance with assign_public_ip=false

### ISSUE THREAD

**Issue by stephenpearson** [terraform-providers/terraform-provider-oci#593]
### Terraform Version

Terraform 0.11.8

### OCI Provider Version

terraform-provider-oci 2.2.4

### Description:

If an oci_core_instance is created with "create_vnic_details.assign_public_ip = false" then if the resource is removed from the statefile it cannot be imported without Terraform destroying and recreating it during the next apply.  This does not occur if the configuration specifies "assign_public_ip = true".

Assume we're using the configuration below and have reached a stable state so that `terraform apply` does nothing.  I then run `terraform state rm oci_core_instance.server` followed by `terraform import oci_core_instance.server ocid1.instance.oc1.phx........` and finally `terraform apply` again.

```
Terraform will perform the following actions:

-/+ oci_core_instance.server (new resource required)
      id:                                           "ocid1.instance.oc1.phx....." => <computed> (forces new resource)
...<snipped>...
      create_vnic_details.0.assign_public_ip:       "true" => "false" (forces new resource)
```

It appears that the provider is not probing the state of "assign_public_ip" from OCI correctly.

### Terraform Plan

Example:
```
resource "oci_core_instance" "server" {
  availability_domain = "niat:PHX-AD-1"
  compartment_id = "ocid1.compartment.oc1..........."
  display_name = "bugtest"
  shape = "VM.Standard1.2"

  source_details {
    source_type = "image"
    source_id   = "ocid1.image.oc1.phx........."
  }

  create_vnic_details {
    subnet_id = "ocid1.subnet.oc1.phx.........."
    display_name = "bugtest"
    hostname_label = "bugtest"
    assign_public_ip = false
  }
}
```

**Comment by alexng-canuck** [comment]
@stephenpearson Thanks for bringing this to our attention.

This appears to be caused by the design of GetInstance API; which does not return the assign_public_ip attribute even though it is supported by the LaunchInstance API.

We are investigating a solution for this in the provider. One possibility is to check whether there's a proper value for the instance public_ip before setting this attribute.

**Comment by stephenpearson** [comment]
Yes you're right, that attribute is indeed missing from getinstance.  Your proposed fix sounds like it would work though; Thanks for looking at this.

**Comment by alexng-canuck** [comment]
@stephenpearson , this issue has been fixed in the latest v3.2.0.

Closing this bug. Please reopen this bug if v3.2.0 doesn't solve the issue.
